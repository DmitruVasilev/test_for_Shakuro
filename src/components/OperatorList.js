import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Operators = styled.ul`
  margin-top: 20px;
`;

const OperatorItem = styled.li`
  display: flex;
  align-items: center;
  border-radius: 5px;
  border: 0.5px solid #d7d7d7;
  margin-bottom: 10px;
  padding: 10px;
`;

const OperatorTitle = styled.h3`
  margin: 0 0 0 10px;
  font-weight: normal;
  font-size: 20px;
  line-height: 1;
`;

const OperatorLink = styled(Link)`
  margin-left: auto;
  padding: 5px 20px;
  text-decoration: none;
  border: 0.5px solid #3ac831;
  color: #3ac831;
  border-radius: 5px;
  background: white;
  cursor: pointer;
  transition: 0.5s ease;
  &:hover,
  &:focus {
    background: rgba(58, 200, 49, 0.15);
  }
`;

const OperatorList = ({ operators }) => (
  <Operators>
    {operators.map(operator => (
      <OperatorItem key={operator.id}>
        <img src={operator.icon_url} alt={operator.title} />
        <OperatorTitle>{operator.title}</OperatorTitle>
        <OperatorLink to={`/pay_mobile/${operator.id}`}>Оплатить</OperatorLink>
      </OperatorItem>
    ))}
  </Operators>
);

OperatorList.propTypes = {
  operators: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      icon_url: PropTypes.string,
      title: PropTypes.string
    })
  ).isRequired
};

export default OperatorList;
