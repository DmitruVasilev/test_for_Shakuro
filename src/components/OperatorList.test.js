/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import OperatorList from "./OperatorList";

const operatorsMock = {
  operators: [
    {
      id: "1",
      title: "MTS",
      icon_url: "https://citycard.ru/i/landing-features/operator-mts.svg"
    },
    {
      id: "2",
      title: "Megafon",
      icon_url: "https://citycard.ru/i/landing-features/operator-megafon.svg"
    },
    {
      id: "3",
      title: "Beeline",
      icon_url: "https://citycard.ru/i/landing-features/beeline.svg"
    }
  ]
};

describe("<OperatorList />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <OperatorList {...operatorsMock} />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <OperatorList {...operatorsMock} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
