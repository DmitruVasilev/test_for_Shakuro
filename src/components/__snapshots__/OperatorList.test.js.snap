// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperatorList /> renders correctly 1`] = `
<ul
  className="sc-bdVaJa fQzoeR"
>
  <li
    className="sc-bwzfXH ixeKvC"
  >
    <img
      alt="MTS"
      src="https://citycard.ru/i/landing-features/operator-mts.svg"
    />
    <h3
      className="sc-htpNat gsjPjQ"
    >
      MTS
    </h3>
    <a
      className="sc-bxivhb jnXvEC"
      href="/pay_mobile/1"
      onClick={[Function]}
    >
      Оплатить
    </a>
  </li>
  <li
    className="sc-bwzfXH ixeKvC"
  >
    <img
      alt="Megafon"
      src="https://citycard.ru/i/landing-features/operator-megafon.svg"
    />
    <h3
      className="sc-htpNat gsjPjQ"
    >
      Megafon
    </h3>
    <a
      className="sc-bxivhb jnXvEC"
      href="/pay_mobile/2"
      onClick={[Function]}
    >
      Оплатить
    </a>
  </li>
  <li
    className="sc-bwzfXH ixeKvC"
  >
    <img
      alt="Beeline"
      src="https://citycard.ru/i/landing-features/beeline.svg"
    />
    <h3
      className="sc-htpNat gsjPjQ"
    >
      Beeline
    </h3>
    <a
      className="sc-bxivhb jnXvEC"
      href="/pay_mobile/3"
      onClick={[Function]}
    >
      Оплатить
    </a>
  </li>
</ul>
`;
